src/services/igc-service.js

```diff
@@ -253,7 +261,13 @@ const igcService = {
    * @return {number} – Returns altitude in meters.
    */
   getAltitudeFromBRecord(BRecord) {
-    return Number(BRecord.substr(25, 5));
+    let altitude = Number(BRecord.substr(25, 5));
+    if (altitude === 0) {
+      // fall back to GPS data
+      // ... well, for real pressure altitude of '0', it will be wrong but this should not happen too often ;-)
+      altitude = Number(BRecord.substr(30, 5));
+    }
+    return altitude;
```

Comment:
I'm curious, do you have IGC tracks with pressure altitude reported as zeros?

In old IGC format gps altitude is reccommended and not mandatory, so some devices may not have it in `B` records. No need to worry, I will test on my old IGC files and make code changes if needed.

Reply:
Yes, it is zero with what I currently use: XCTrack on my mobile phone (without barometer sensor)  ... actually just for convenience to easily add my flight to my training log book (and koifly logs :-)).

here is a B entry:
B1806225218422N01305423EA000000005348

I can send you a full igc file, if you like.

Reply:
Interesting... Yes, could you please send me your file, I want to see if there's any indication that a device doesn't have a barometer.